src/main.js

```diff
@@ -33,55 +28,80 @@ process.on('uncaughtException', (exception: Error) => {
  *
  * See https://github.com/justmoon/constitute for more info
  */
-const container = new Container();
-defaultBindings(container, settings);
 
-const deviceServer = container.constitute('DeviceServer');
-deviceServer.start();
+const startServers = () => {
+  const container = new Container();
+  defaultBindings(container, settings);
+  const deviceServer = container.constitute('DeviceServer');
+  const app = createApp(container, settings);
 
-const app = createApp(container, settings);
+  deviceServer.start();
 
-const onServerStartListen = (): void =>
-  console.log(`express server started on port ${NODE_PORT}`);
+  const onServerStartListen = (): void =>
+    console.log(`express server started on port ${NODE_PORT}`);
 
-const {
-  SSL_PRIVATE_KEY_FILEPATH: privateKeyFilePath,
-  SSL_CERTIFICATE_FILEPATH: certificateFilePath,
-  USE_SSL: useSSL,
-  ...expressConfig
-} = settings.EXPRESS_SERVER_CONFIG;
+  const {
+    SSL_PRIVATE_KEY_FILEPATH: privateKeyFilePath,
+    SSL_CERTIFICATE_FILEPATH: certificateFilePath,
+    USE_SSL: useSSL,
+    ...expressConfig
+  } = settings.EXPRESS_SERVER_CONFIG;
 
-if (useSSL) {
-  const options = {
-    cert: certificateFilePath && fs.readFileSync(
-      nulltrhows(certificateFilePath),
-    ),
-    key: privateKeyFilePath && fs.readFileSync(
-      nulltrhows(privateKeyFilePath),
-    ),
-    ...expressConfig,
-  };
-  https
-    .createServer(options, (app: any))
-    .listen(NODE_PORT, onServerStartListen);
+  if (useSSL) {
+    const options = {
+      cert: certificateFilePath && fs.readFileSync(
+        nulltrhows(certificateFilePath),
+      ),
+      key: privateKeyFilePath && fs.readFileSync(
+        nulltrhows(privateKeyFilePath),
+      ),
+      ...expressConfig,
+    };
+    https
+      .createServer(options, (app: any))
+      .listen(NODE_PORT, onServerStartListen);
+  } else {
+    http
+      .createServer((app: any))
+      .listen(NODE_PORT, onServerStartListen);
+  }
+
+  process.on('uncaughtException', (exception: Error) => {
```

Comment:
forks restarts should be handled automatically by express-cluster: https://github.com/Flipboard/express-cluster#config `respawn` in config